admin_ui/src/components/MediaViewer.vue

```diff
@@ -0,0 +1,213 @@
+<template>
+    <div id="media_viewer">
+        <!-- Top bar -->
+
+        <div class="top_bar">
+            <p class="file_key">
+                <span>File:</span> {{ mediaViewerConfig.fileKey }}
+            </p>
+
+            <p class="close">
+                <a href="#" @click.prevent="$emit('close')"
+                    ><font-awesome-icon icon="times"></font-awesome-icon
+                ></a>
+            </p>
+        </div>
+
+        <!-- Media containers -->
+
+        <div
+            v-if="isImage(mediaViewerConfig.fileKey)"
+            class="image_container"
+            v-bind:style="{ backgroundImage: `url(${fileURL})` }"
+        ></div>
+
+        <div
+            v-else-if="isVideo(mediaViewerConfig.fileKey)"
+            class="video_container"
+        >
+            <video controls :src="fileURL">Video playback not available</video>
+        </div>
+
+        <div class="no_preview" v-else>
+            <p>
+                <font-awesome-icon icon="file"></font-awesome-icon> Unable to
+                preview file - download below.
+            </p>
+        </div>
+
+        <!-- Bottom bar -->
+
+        <div class="bottom_bar">
+            <p>
+                <a class="button" :href="fileURL" download
+                    ><font-awesome-icon icon="download"></font-awesome-icon>
+                    Download</a
+                >
+            </p>
+        </div>
+    </div>
+</template>
```

Comment:
This is great. I really like it. Looks cool.

Reply:
Thanks!